src/interpreter/vm.cc

```diff
@@ -12,30 +12,12 @@ namespace verona::interpreter
   void VM::run(std::vector<Value> args, size_t cown_count, size_t start)
   {
     halt_ = false;
-    start_ip_ = ip_ = start;
+    push_frame(start, 0, OnReturn::Halt);
 
-    std::string_view name = code_.str(ip_);
-
-    frame_ = Frame::initial();
-    frame_.argc = code_.u8(ip_);
-    frame_.retc = code_.u8(ip_);
-    frame_.locals = code_.u8(ip_);
-    code_.u32(ip_); // size
-
-    assert(static_cast<size_t>(frame_.argc) == args.size());
-
-    trace(
-      "Entering function {}, argc={:d} retc={:d} locals={:d}",
-      name,
-      frame_.argc,
-      frame_.retc,
-      frame_.locals);
+    assert(static_cast<size_t>(frame().argc) == args.size());
 
     // Load registers with cowns values.
```

Comment:
Is this comment in the right place?

Reply:
I've removed it, since it was pretty out-of-date I think, and expanded the one below a bit.